bin/check_everything.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+"""
+This script verifies that the Submitty installation and course data
+directories are configured and permissioned correctly.
+
+This script is a work-in-progress.  Additional checks will be added or
+revised as needed.
+"""
+
+import os
+import pwd
+import stat
+import grp
+import sys
+
+
+# Will be filled in by INSTALL_SUBMITTY.sh
+SUBMITTY_INSTALL_DIR="__INSTALL__FILLIN__SUBMITTY_INSTALL_DIR__"
+SUBMITTY_DATA_DIR="__INSTALL__FILLIN__SUBMITTY_DATA_DIR__"
+
+HWPHP_USER="__INSTALL__FILLIN__HWPHP_USER__"
+HWCRON_USER="__INSTALL__FILLIN__HWCRON_USER__"
+
+COURSE_BUILDERS_GROUP="__INSTALL__FILLIN__COURSE_BUILDERS_GROUP__"
+
+###########################################################################
+def CheckItemBits (my_path, is_dir, my_owner, my_group, my_bits):
+    ret_val = True
+    try:
+        pwd.getpwnam(my_owner)
+    except KeyError:
+        sys.stderr.write("ERROR! user "+my_owner+" does not exist\n")
+        ret_val = False
+    try:
+        grp.getgrnam(my_group)
+    except KeyError:
+        sys.stderr.write("ERROR! group "+my_group+" does not exist\n")
+        ret_val = False
+    if is_dir and not os.path.isdir(my_path):
+        sys.stderr.write("ERROR! "+my_path+" should be a directory!\n")
+        ret_val = False
+    if not is_dir and os.path.isdir(my_path):
+        sys.stderr.write("ERROR! "+my_path+" should not be a directory!\n")
+        ret_val = False
+    if not os.path.exists(my_path):
+        sys.stderr.write("ERROR! "+my_path+" does not exist\n")
+        ret_val = False
+    if pwd.getpwuid(os.stat(my_path).st_uid).pw_name != my_owner:
+        sys.stderr.write("ERROR! "+my_path+" should be owned by "+my_owner+"\n")
+        ret_val = False
+    if grp.getgrgid(os.stat(my_path).st_gid).gr_name != my_group:
+        sys.stderr.write("ERROR! "+my_path+" should be group "+my_group+"\n")
+        ret_val = False
+    bits = os.stat(my_path)[stat.ST_MODE]
+    bits &= 0o777
+    if bits != my_bits:
+        sys.stderr.write("ERROR! "+my_path+" permission is "+oct(bits)+" should be "+oct(my_bits)+"\n")
+        ret_val = False
+    return ret_val
+
+
+###########################################################################
+def CheckCourseInstructorAndGroup(my_instructor, my_group):
+    ret_val = True
+    try:
+        pwd.getpwnam(my_instructor)
+    except KeyError:
+        sys.stderr.write("ERROR! user "+my_instructor+" does not exist\n")
+        ret_val = False
+    try:
+        grp.getgrnam(my_group)
+    except KeyError:
+        sys.stderr.write("ERROR! group "+my_group+" does not exist\n")
+        ret_val = False
+    c_g = grp.getgrnam(my_group)
```

Comment:
This will throw a KeyError if `my_group` does not eixst.

I think it'd make sense to do:
```
cb_g = grp.getgrnam(COURSE_BUILDERS_GROUP)
try:
    c_g = grp.getgrname(my_group)
    try:
        pwd.getpwnam(my_instructor)
        if not my_instructor in c_g.gr_mem:
            ...
        if not my_instructor in cb_g.gr_mem:
            ...
    except KeyError:
        ...
    if not HWPHP_USER in c_g.gr_mem:
        ...
    if not HWCRON_USER in c_g.gr_mem:
        ...
except KeyError:
    ...
```

as no point doing the c_g checks if the group doesn't exist and same with the checks for instructor if it doesn't exist.